### PR TITLE
DAOS-13493 placement: only check spare target if necessary

### PR DESCRIPTION
**Pull request by wangdi1**
Only check spare target if layout version > 0, since before 2.4, it will not assign the shard if there are no spare targets.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/placement/jump_map.c [hunk 1]
```diff
@@ -341,9 +341,18 @@ obj_remap_shards(struct pl_jump_map *jmap, uint32_t layout_ver, struct daos_obj_
 			get_target(root, layout_ver, &spare_tgt, crc(key, rebuild_key),
 				   dom_used, dom_full, dgu->dgu_used, tgts_used, shard_id,
 				   allow_status, fdom_lvl, &spares_left, &spare_avail);
-			D_ASSERT(spare_tgt != NULL);
-			D_DEBUG(DB_PL, "Trying new target: "DF_TARGET"\n",
-				DP_TARGET(spare_tgt));
+			if (layout_ver > 1) {
```
Comment (ryon-jensen) [inline src/placement/jump_map.c:344]: Your commit message says "only check ... layout version > 0", but here the check is "layout_ver > 1"?

**Review by ryon-jensen — APPROVED** [review]
Looks good to me

File: src/placement/jump_map.c (hunk 1 above)
Comment (wangdi1) [reply src/placement/jump_map.c:344]: ah, this should > 0. Thanks!

**Review by liuxuezhao — APPROVED** [review]

**Review by liuxuezhao — APPROVED** [review]

**Comment by wangdi1** [comment]
ping

**Comment by wangdi1** [comment]
ping

**Review by ryon-jensen — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**